### ISSUE TITLE
Should setting a custom executable for PHPUnit disable the `--colors` option?

### ISSUE THREAD

**Issue by bigsweater** [vim-test/vim-test#670]
Hey there! Thanks for this awesome plugin.

I have the following in my `.vim/ftplugin/php.vim`:

```vim
function! s:get_runner() abort
    if filereadable('./artisan') && isdirectory('./vendor/nunomaduro/collision')
        return 'php artisan test'
    endif

    return test#php#phpunit#executable()
endfunction

function! s:get_options() abort
    if filereadable('./vendor/bin/paratest') && (g:test#php#phpunit#executable == 'php artisan test')
        return { 'suite': '--parallel' }
    endif

    return {}
endfunction

let test#php#phpunit#executable = s:get_runner()
let test#php#phpunit#options = s:get_options()
```

This only sets the `phpunit` executable to `php artisan test` if `artisan` is available and `nunomaduro/collision` is installed. It also adds a `--parallel` option if paratest is installed and the executable is `php artisan test`.

However, when I run the test suite using the `neovim` strategy, I get the following error:

```sh
PHPUnit 9.5.21 #StandWithUkraine

Unknown option "--parallel"

[Process exited 1]
```

This is because vim-test is still passing the `--colors` option to the `php artisan test` executable, which is tripping up the `artisan` command. You can the issue in the Neovim terminal window info:

`TERMINAL  term://~/Code/{my project name}//21772:php artisan test --colors --parallel`

I confirmed this error by typing `php artisan test --colors --parallel` into a terminal window in my project directory.

(Oddly, it works fine if the command is `php artisan test --parallel --colors` -- might be a bug in artisan.)

The PHPUnit runner *always* passes `--colors`, no matter what:

https://github.com/vim-test/vim-test/blob/206eb5492b4afa03c31cea529aa917b4a7085d10/autoload/test/php/phpunit.vim#L39-L47

I think it's arguable that if we're using a custom executable, *no* options should be passed by default. 

Is there a way to override this behavior? Or would a PR be welcome to check for the global executable, before adding default options? (I understand this check would need to happen in every PHP test runner & might have backwards compatibility issues.)

**Comment by codeinabox** [comment]
> I think it's arguable that if we're using a custom executable, _no_ options should be passed by default.
> 
> Is there a way to override this behavior? Or would a PR be welcome to check for the global executable, before adding default options? (I understand this check would need to happen in every PHP test runner & might have backwards compatibility issues.)

Thank you! When you say no options at all, do you mean the `colors` option or any options? The former is part of the [PHPUnit command line options](https://docs.phpunit.de/en/10.3/textui.html), so I'd argue that any custom executable should support any of these options

**Comment by bigsweater** [comment]
Honestly it's been awhile since I looked at this code, and I think I abandoned this particular config! 

My thinking was *any* options at all -- the idea being that if a user wants to use a custom executable, they would also want to customize the options passed to it, and may not always want e.g. `--color` as part of their test command.

That said -- I'll close this! I can't even remember the fix I had in mind, if any. Thanks for taking a look.
